### PR TITLE
Data provider should check the entire POM file

### LINKED ISSUE #234: Data providers should check the entire pom.xml
The report for Apache Flink says that it doesn't use OWASP Dependency Check

https://github.com/SAP/fosstars-rating-core/blob/master/docs/oss/security/apache/flink.md - it says No but the project uses the plugin

but the project actually uses it:

https://github.com/apache/flink/blob/master/pom.xml#L2015

Looks like the data providers can't find it for some reason. Maybe they ignore pluginManagement section and profiles.

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of main updates:

- Updated the existing data providers to check profiles and plugin management sections in POM files.
- Updated the tests.

This closes #234

### CODE REVIEW

**Comment by sourabhsparkala** [comment]
However, after the discussion regarding the possible introduction of a Helper class which can be used by all the Data Providers when parsing `pom.xml` plugins and configurations.

**Comment by artem-smotrakov** [comment]
> However, after the discussion lets introduce a Helper class that can be used by all the Data Providers when parsing `pom.xml` plugins and configurations.

@sourabhsparkala I've implemented visitors for POM files. Please have a look.

File: src/main/java/com/sap/sgs/phosphor/fosstars/maven/AbstractModelVisitor.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.sap.sgs.phosphor.fosstars.maven;
+
+import java.util.Set;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.ReportPlugin;
+
+/**
+ * An implementation of {@link ModelVisitor} that doesn't nothing.
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/maven/AbstractModelVisitor.java:8]: typo

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyCheck.java [hunk 2]
```diff
@@ -213,4 +117,36 @@ private static boolean isDependencyCheck(ReportPlugin plugin) {
...
+
+  /**
+   * Creates a new visitor for searching for OWASP Dependency Check in a POM file.
+   */
+  private static Visitor withVisitor() {
+    return new Visitor();
+  }
+
+  /**
+   * A visitor for searching for OWASP Dependency Check in a POM file.
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyCheck.java:129]: A visitor for searching OWASP Dependency Check?

File: src/test/java/com/sap/sgs/phosphor/fosstars/data/github/UsesNoHttpToolTest.java [hunk 3]
```diff
@@ -58,14 +56,11 @@ public void testGradleWithoutNoHTTP() throws IOException {
...
-    when(content.read()).thenReturn(is);
+    final LocalRepository repository = mock(LocalRepository.class);
+    when(repository.read(filename)).thenReturn(Optional.of(is));
 
-    GHRepository repository = mock(GHRepository.class);
-    when(repository.getFileContent(filename)).thenReturn(content);
-
-    when(fetcher.github().getRepository(any())).thenReturn(repository);
+    GitHubProject project = new GitHubProject("org", "test");
+    fetcher.addForTesting(project, repository);
```
Comment (sourabhsparkala) [inline src/test/java/com/sap/sgs/phosphor/fosstars/data/github/UsesNoHttpToolTest.java:63]: Is there no test case to check where `LOCATION` it was expected to be found?

**Comment by sourabhsparkala** [comment]
@artem-smotrakov minor comments and questions.

File: src/test/java/com/sap/sgs/phosphor/fosstars/data/github/UsesNoHttpToolTest.java (hunk 3 above)
Comment (artem-smotrakov) [reply src/test/java/com/sap/sgs/phosphor/fosstars/data/github/UsesNoHttpToolTest.java:63]: Currently there is no tests for that. In fact, locations are not used at all here. They are necessary for #232. The test will be added there. However, I can also add tests for the visitors that use the existing POM files.

**Comment by artem-smotrakov** [comment]
I've added a test. Thanks for the review and comments @sourabhsparkala !

**Pull request merged**